pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -149,19 +150,26 @@ func (a *v1beta1PolicyApplier) AuthNFilter(forSidecar bool) *http_conn.HttpFilte
 func (a *v1beta1PolicyApplier) InboundMTLSSettings(endpointPort uint32, node *model.Proxy, trustDomainAliases []string) plugin.MTLSSettings {
 	effectiveMTLSMode := a.GetMutualTLSModeForPort(endpointPort)
 	authnLog.Debugf("InboundFilterChain: build inbound filter change for %v:%d in %s mode", node.ID, endpointPort, effectiveMTLSMode)
+	var mc *meshconfig.MeshConfig
+	if a.push != nil {
```

Comment:
nit: Could even simplify to call getMinTLSVersion() here (Only need to call once) and then pass the result `TlsParameters_TlsProtocol` directly to BuildInboundTLS().